detector/src/main/java/com/synopsys/integration/detector/result/DetectorResult.java

```diff
@@ -42,4 +44,27 @@ public boolean getPassed() {
     public String getDescription() {
         return description;
     }
+
+    public String getStatusCode() {
+        if (!passed) {
+            return formatResultNameToStatusCode(resultName);
+        }
+        return "PASSED";
+    }
+
+    private String formatResultNameToStatusCode(String resultName) {
```

Comment:
I think this should be done before this class is constructed - ie, this class is given the formatted result name.

Reply:
My concern with that proposal is that then the format method would need to become a class, and all the places that instantiate a DetectorResult would need to instantiate that class.  It just seems messier than this setup where DetectorResult is passed the name of the result it is actually wrapping, and then determines its own status code internally, making the formatting happen in one place.

Reply:
I'd like to keep these kind of result classes as simple data objects and immutable. I also don't see any tests for this code which would be easier to test if it lived in its own formatter class. And I think tests for this are important as we don't want them to accidentally change. I also prefer that in general, getters should be as simple as possible.

What I would be inclined to do is have Detect Result only contain the raw class name (we could call it ResultClassName) and formatted output manager does the conversion to display name (formatted name) - I am inclined to think if we dramatically changed the format of the names, we would have to bump the formatted output version so I'd also prefer the conversion from some internal class name to a formal external value be as close to formatted output manager as possible.

Regardless, I think that I would have preferred a DetectResultFactory which enforced the formatting (and the getter was still simple) to doing the work in the getter.

Reply:
@taikuukaits I did some refactoring to move the conversion of result class name to FormattedOutputManager.  I also added a test.